### PR TITLE
support multi-platform build

### PR DESCRIPTION
**Pull request by justadogistaken**

### CODE REVIEW

**Comment by kakaZhou719** [comment]
for manifest object:
sealer manifest create localhost/my-manifest
sealer manifest add localhost/my-manifest localhost/ack-distro-amd64:v1
sealer manifest inspect localhost/my-manifest
sealer manifest push localhost/my-manifest docker.io/sealer/my-kubernetes:1.19.8  --all
sealer manifest remove localhost/my-manifest sha256:be14feecd1103bdb28db7f7d09ad3ee56f7afc0bb4ddf5f0284e9dc0158e9a49
sealer manifest delete localhost/my-manifest


for mutil-arch images:
sealer build -f Kubefile --manifest my-kubernetes:1.19.8 --platform linux/amd64,linux/arm64
sealer manifest push localhost/my-kubernetes:1.19.8 docker.io/sealer/my-kubernetes:1.19.8  --all
sealer pull docker.io/sealer/my-kubernetes:1.19.8 --platform linux/arm64

**Comment by kakaZhou719** [comment]
for mutil-arch image:
sealer build -f Kubefile --manifest my-kubernetes:1.19.8 --platform linux/amd64,linux/arm64
sealer manifest push localhost/my-kubernetes:1.19.8 docker.io/sealer/my-kubernetes:1.19.8  --all: push all platform image
sealer manifest push localhost/my-kubernetes:1.19.8 docker.io/sealer/my-kubernetes:1.19.8: push default platform image
sealer pull docker.io/sealer/my-kubernetes:1.19.8 --platform linux/arm64


for single-arch image:
sealer build -f Kubefile -t my-kubernetes:1.19.8
sealer push my-kubernetes:1.19.8
sealer pull docker.io/sealer/my-kubernetes:1.19.8

**Comment by Starnop** [comment]
@justadogistaken cloud you please rebase the commits for this PR

image

**Comment by kakaZhou719** [comment]
> @justadogistaken cloud you please rebase the commits for this PR
> 
> image

@Starnop ,let me do it， some rebase commits was pushed by me.

File: pkg/imageengine/buildah/manifest.go [hunk 1]
```diff
@@ -15,29 +15,177 @@
...
+func (engine *Engine) DeleteManifests(names []string, opts *options.ManifestDeleteOpts) error {
+	runtime := engine.ImageRuntime()
+
+	rmiReports, rmiErrors := runtime.RemoveImages(context.Background(), names, &libimage.RemoveImagesOptions{
+		Filters:        []string{"readonly=false"},
+		LookupManifest: true,
+	})
+	for _, r := range rmiReports {
+		for _, u := range r.Untagged {
+			fmt.Printf("untagged: %s\n", u)
```
Comment (Starnop) [inline pkg/imageengine/buildah/manifest.go:72]: try to encapsulate a method to printf with a newline

File: pkg/imageengine/buildah/manifest.go [hunk 2]
```diff
@@ -15,29 +15,177 @@
...
+	appendErr := func(e error) {
+		if latestErr == nil {
+			latestErr = e
+		} else {
+			latestErr = fmt.Errorf("tried %v: %w", e, latestErr)
+		}
+	}
+
+	for _, ref := range refs {
+		logrus.Debugf("Testing reference %q for possible manifest", transports.ImageName(ref))
```
Comment (Starnop) [inline pkg/imageengine/buildah/manifest.go:154]: I don't know what our principles for using `logrus` and `fmt` are?

Could you please make a simple explanation?

File: pkg/imageengine/buildah/commit.go [hunk 3]
```diff
@@ -31,57 +31,60 @@ import (
...
 
 	// If the user specified an image, we may need to massage it a bit if
 	// no transport is specified.
 	// TODO we support commit to local image only, we'd better limit the input of name
-	if dest, err = alltransports.ParseImageName(image); err != nil {
-		candidates, err := shortnames.ResolveLocally(systemCxt, image)
-		if err != nil {
-			return err
+	if image != "" {
+		if dest, err = alltransports.ParseImageName(image); err != nil {
```
Comment (Starnop) [inline pkg/imageengine/buildah/commit.go:69]: ugly code here, how about using a single method to parse the image name?

File: pkg/imageengine/buildah/build.go [hunk 4]
```diff
@@ -340,10 +340,11 @@ func (engine *Engine) migrateFlags2Wrapper(opts *options.BuildOptions, wrapper *
 	flags := engine.Flags()
 	// imageengine cache related flags
 	// cache is enabled when "layers" is true & "no-cache" is false
```
Comment (Starnop) [inline pkg/imageengine/buildah/build.go:342]: not matched comment with code logic

File: pkg/define/options/options.go [hunk 5]
```diff
@@ -161,3 +165,21 @@ type TagOptions struct {
...
+}
+
+type ManifestInspectOpts struct {
+}
+
+type ManifestDeleteOpts struct {
+}
+
+type ManifestAddOpts struct {
+	Os, Arch, Variant, OsVersion string
```
Comment (Starnop) [inline pkg/define/options/options.go:179]: unfold them, not a good style In my opinion

**Comment by kakaZhou719** [comment]
for manifest object:
sealer manifest create localhost/my-manifest
sealer manifest add localhost/my-manifest localhost/ack-distro-amd64:v1
sealer manifest inspect localhost/my-manifest
sealer manifest push localhost/my-manifest
sealer manifest remove localhost/my-manifest sha256:be14feecd1103bdb28db7f7d09ad3ee56f7afc0bb4ddf5f0284e9dc0158e9a49
sealer manifest delete localhost/my-manifest


for mutil-arch image:
sealer build -f Kubefile -t docker.io/sealer/my-kubernetes:1.19.8 --platform linux/amd64,linux/arm64
sealer push docker.io/sealer/my-kubernetes:1.19.8 --all: push all platform image
sealer push docker.io/sealer/my-kubernetes:1.19.8: push default platform image
sealer pull docker.io/sealer/my-kubernetes:1.19.8 --platform linux/arm64

for single-arch image:
sealer build -f Kubefile -t my-kubernetes:1.19.8
sealer push my-kubernetes:1.19.8
sealer pull docker.io/sealer/my-kubernetes:1.19.8

**Review by Starnop — APPROVED** [review]

**Pull request merged**